vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java

```diff
@@ -2059,6 +2074,345 @@ private void createNotBufferedDynamicEditor() {
+                    .getDataProvider();
+            List<Person> sourceItems = new ArrayList<>(
+                    sourceDataProvider.getItems());
+            sourceItems.removeAll(draggedItems);
+            dragSource.setItems(sourceItems);
+
+            // Add dragged items to the target Grid
+            Grid<Person> targetGrid = event.getSource();
+            ListDataProvider<Person> targetDataProvider = (ListDataProvider<Person>) targetGrid
+                    .getDataProvider();
+            List<Person> targetItems = new ArrayList<>(
+                    targetDataProvider.getItems());
+
+            int index = target.map(person -> targetItems.indexOf(person)
+                    + (event.getDropLocation() == DropLocation.BELOW ? 1 : 0))
+                    .orElse(0);
+            targetItems.addAll(index, draggedItems);
+            targetGrid.setItems(targetItems);
+        };
+
+        PersonService personService = new PersonService();
+
+        grid.setItems(personService.fetch(0, 5));
+        grid.setSelectionMode(Grid.SelectionMode.MULTI);
+        grid.addDropListener(dropListener);
+        grid.addDragStartListener(dragStartListener);
+        grid.addDragEndListener(dragEndListener);
+        grid.setRowsDraggable(true);
+        grid.setColumns("firstName", "lastName");
+
+        grid2.setItems(personService.fetch(5, 1));
+        grid2.setSelectionMode(Grid.SelectionMode.MULTI);
+        grid2.addDropListener(dropListener);
+        grid2.addDragStartListener(dragStartListener);
+        grid2.addDragEndListener(dragEndListener);
+        grid2.setRowsDraggable(true);
+        grid2.setColumns("firstName", "lastName");
+
+        // end-source-example
+
+        HorizontalLayout hl = new HorizontalLayout(grid, grid2);
+
+        addCard("Drag and Drop", "Drag Rows Between Grids", hl);
+    }
+
+    private void createDropLocations() {
+
+        // begin-source-example
+        // source-example-heading: Drop Location
+        PersonService personService = new PersonService();
+
+        TreeGrid<Person> treeGrid = new TreeGrid<>();
+        Grid<Person> grid = new Grid<>(Person.class);
+        grid.setSelectionMode(SelectionMode.NONE);
+        grid.setItems(personService.fetch(0, 50));
+
+        grid.addDragStartListener(event -> {
+            draggedItem = event.getDraggedItems().get(0);
+            treeGrid.setDropMode(DropMode.ON_TOP_OR_BETWEEN);
+        });
+
+        grid.addDragEndListener(event -> {
+            draggedItem = null;
+            treeGrid.setDropMode(null);
+        });
+        grid.setColumns("firstName", "lastName", "phoneNumber");
+        grid.setRowsDraggable(true);
+
+        TreeData<Person> td = new TreeData<>();
+        td.addItems(null, personService.fetch(51, 2));
+
+        treeGrid.setDataProvider(new TreeDataProvider<Person>(td));
+        /*
+         * This enables dropping in between or on top of the existing grid rows.
```

Comment:
This comment is now out of place after moving the `setDropMode` to drag start listener

Reply:
Done.